src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -0,0 +1,458 @@
+        """
+        Inactive pileup task:
+
+        This task is supposed to look at pileup documents that have been set to
+        inactive. The main goal here is to ensure that there are no Rucio rules
+        left in the system (of course, for the relevant DID and the Rucio
+        account adopted by our microservice). Pileup documents that are updated
+        as a result of this logic should have their data persisted back in
+        MongoDB. A short algorithm for it can be described as follows:
+
+        1. Read pileup document from MongoDB with filter active=false
+        2. for each DID and Rucio account, get a list of all the existent rules
+           - make a Rucio call to delete that rule id, then:
+             - remove the rule id from ruleIds (if any) and remove the RSE name from currentRSEs (if any)
+        3. make a log record if the DID + Rucio account tuple does not have any existent rules
+           - and set ruleIds and currentRSEs to an empty list
+        4. once all the relevant rules have been removed, persist an up-to-date version of the pileup data structure in MongoDB
+        """
+        spec = {'active': True}
+        docs = self.mgr.getPileup(spec)
+        taskSpec = self.getTaskSpec()
+        asyncio.run(performTasks('inactive', docs, taskSpec))
+
+    def activeTask(self, marginSpace=102**3):
+        """
+        Active pileup task:
+        :param marginSpace: minimum margin space size in bytes to have at RSE to place a dataset
+
+        This task is supposed to look at pileup documents active in the system.
+        Its main goal is to ensure that the pileup DID has all the requested
+        rules (and nothing beyond them), according to the pileup object
+        configuration. Pileup documents that are updated as a result of this
+        logic should have their data persisted back in MongoDB.
+
+        1. Read pileup document from MongoDB with filter active=true
+        2. if expectedRSEs is different than currentRSEs, then further data placement is required (it's possible that data removal is required!)
+        3. for each expectedRSEs not in currentRSEs:
+           - check if there is an ongoing Rucio rule for that DID + Rucio account + RSE (we might want to remove RSE from this call?)
+           - if there is a Rucio rule, then ensure that it's listed under ruleIds, otherwise add it
+           - if there is none, then a new rule needs to be created. First, check whether the RSE has enough space available for that
+             - if it does, create the rule and append the rule id to ruleIds
+             - otherwise, make a log record saying that there is not enough space
+        4. once all the relevant rules have been created, persist an up-to-date version of the pileup data structure in MongoDB
+
+        """
+        spec = {'active': True}
+        docs = self.mgr.getPileup(spec)
+        taskSpec = self.getTaskSpec()
+        taskSpec['task'] = 'active'
```

Comment:
why is this `task` needed here?

Reply:
it is left-over from several iteration of code development, I'm removing it.